### PR TITLE
Feature/eng 00012 impoer expeort stack

### PR DESCRIPTION
**Pull request by bayoumi17m**
# Overview
Added the ability to now export and import stack components along with entire stacks via the CLI.

## Changes
- Added new `stack export` and `stack import` commands

## API Updates
- CLI has new commands for stack group
    - `stack export` generates YAML on stack configuration
    - `stack import` loads YAML to create a stack

## Checklist
- [ ] Interrogate has > 70% coverage
- [ ] All tests have passed (new tests are made)
- [ ] Total test coverage > 75%
- [ ] New code test coverage > 80%
- [ ] Pass all linting checks

### CODE REVIEW

**Review by JSun14 — APPROVED** [review]
cool beans merge this shit

File: src/coalescenceml/cli/stack.py [hunk 1]
```diff
@@ -1,15 +1,25 @@
-from typing import Optional
```
Comment (fort745) [inline src/coalescenceml/cli/stack.py:1]: image
Container registry column name does not show

File: src/coalescenceml/cli/stack.py [hunk 2]
```diff
@@ -339,3 +374,156 @@ def down_stack(force: bool = False) -> None:
             f"Suspending resources for active stack '{stack_.name}'."
```
Comment (fort745) [inline src/coalescenceml/cli/stack.py:374]: i think this is a typo

File: src/coalescenceml/cli/stack.py [hunk 3]
```diff
@@ -339,3 +374,156 @@ def down_stack(force: bool = False) -> None:
...
         )
         stack_.suspend()
```
Comment (fort745) [inline src/coalescenceml/cli/stack.py:376]: is this a typo?

File: src/coalescenceml/cli/stack.py [hunk 4]
```diff
@@ -339,3 +374,156 @@ def down_stack(force: bool = False) -> None:
...
+@click.argument("stack_name", type=str, required=True)
+@click.argument(
+    "filepath",
+    type=str,
+    required=False,
+)
+def export_stack(stack_name: Optional[str], filepath: Optional[str]) -> None:
+    """Export a stack to a YAML config."""
+
+    # TODO: Dupliate of describe()
```
Comment (fort745) [inline src/coalescenceml/cli/stack.py:405]: was this todo addressed?

File: src/coalescenceml/cli/stack.py [hunk 5]
```diff
@@ -339,3 +374,156 @@ def down_stack(force: bool = False) -> None:
...
+
+    filepath = filepath or f"{stack_name}-config.yaml"
+    yaml_utils.write_yaml(filepath, yaml_data)
+    cli_utils.info(f"Exported stack '{stack_name}' to file '{filepath}'")
+
+
+def import_component_helper(
+    component_flavor: StackComponentFlavor, component_config: Dict[str, str],
+) -> str:
+    """ """
```
Comment (fort745) [inline src/coalescenceml/cli/stack.py:437]: add documentation

**Review by fort745 — APPROVED** [review]
looks good aside from a few minor changes

**Pull request merged**